template/src/api/routes/$$channel$$.js

```diff
@@ -11,9 +13,24 @@ module.exports = router;
   {%- endif %}
 router.use('{{ channelName | toHermesTopic }}', async (message, next) => {
   try {
+    {% if channel.publish().hasMultipleMessages() %}
```

Comment:
the only problem I see is that the generated code is the same for subscribe and publish, except of the operation. You have 2 options:

Option 1. refactor this way that generated code is duplicated (in the end it is generated code so it doesn't matter) but the nunjucks code is also duplicated and should be extracted into a reusable `macro` in `partials` directory - example https://github.com/asyncapi/template-for-generator-templates/blob/nunjucks/partials/listChannels.html
Option 2. refactor this in a way that generated code is not duplicated but is actually using a helper function. I think Option 1 would be easier as writing such helper with nunjucks would add more complexity to template code

Your decision

Reply:
I suggest deferring that refactoring to the future. Both are beyond my skillset. I would rather add some testing for this in some way; I don't actually know how we can validate this change (i.e. test the generated code) to ensure it is working.

Reply:
Yes, that is exactly what I'm saying. The use of `oneOf` in AsyncAPI maps directly to a JSON schema that is defined with `oneOf`, so AsyncAPI should export that schema object for validation purposes.

Reply:
oh, interesting, so `channel.subscribe().message()` would return a `oneOf` JSON Schema?